### PR TITLE
Markdown + YAML for the timeline

### LINKED ISSUE #1235: Refactoring the timeline: YAML + Markdown entries and a slimmer CSS
## The entries

Our [timeline](https://www.pythondiscord.com/timeline/) was added following our 100K members milestone in PR #381. **@<anytag/>ks129** [suggested](https://github.com/python-discord/site/pull/381#issuecomment-703060032) that we should dynamically load the timeline with YAML/Markdown files instead of using raw HTML, however it looks like we went ahead with the latter [due to time constraints](https://github.com/python-discord/site/pull/381#issuecomment-703060326).

@ichard26 lead the project which updated the timeline most recently (#1065). I reached out to see what they think of using YAML/Markdown instead, and they mentioned that updating the timeline again in the future would be a lot easier if it weren't in raw HTML. I propose that we should consider switching to a system similar to our `content` and `resources` app for the timeline.

Here's a general idea of how it could look like. Use the `pydis_site/apps/home/timeline` directory to store markdown files with YAML frontmatter, each file representing a timeline item. The first two could look like this:

`2023-07-11_new-paste-service.md`
```md
---
date: 2023-07-11
icon: fa fa-comments
icon_color: pastel-pink
---

# Switch to new paste service

We migrate over to [pinnwand](https://github.com/supakeen/pinnwand) as the
service that powers our paste bin over at <https://paste.pythondiscord.com/>.
We made the switch as it comes with native light/dark modes, support for
multi-file pastes, additional support for text highlighting languages, and
plus, it's written in Python!
```

`2023-01-30_retirement-of-joe-and-sebastiaan.md`
```md
---
date: 2023-01-30
icon: pydis
---

# Retirement of Joe and Sebastiaan

Having been at the helm of Python Discord for over 5 and 3 years respectively,
Joe and Sebastiaan retire and step down. They gain the **@Founders** role and
continue as advisors to the **@Directors**, the new name of the original
**@Owners** role.

At the same time, Mina and Zig join Leon as co-directors.
```

The file names can be completely arbitrary, because the view would only be looking at the YAML frontmatter and the markdown content when rendering the timeline. The above just shows an example of a file name convention we could adopt, which helps chronological ordering in directory listings.

For the Advent of Code 2020 item's special date format, we could include a key something like `date_label: December 1st - 25th, 2020`. If it's not specified (the case for all other entries) we can use strftime with `date` key.

The files can be read on startup and stored into a data structure that would sort the entries reverse-chronologically. In the view, we pass in the entries as context to a new template file for the timeline. Finally the template would fill in all the entries in HTML.

There are good references for how this can be done in our `content` and `resources` apps.

## The CSS

I tried finding any form of discussion regarding the initial timeline implementation, but failed, maybe it was in core dev channels? I did however notice that the CSS seems to be copied from [this tutorial](https://codeconvey.com/responsive-animated-timeline-javascript-css3/). Despite Joe and other contributors having done some major improvements, there are still quite a number of issues with the CSS that bugs me a lot, most notably:
- Is 3000+ lines of CSS really needed just for this timeline page? We're carrying a 70+ kB baggage of a resource around for this page as extra CSS in addition to the compulsory Bulma and FontAwesome resources.
- Border radius, box shadows, background color are all inconsistent with the rest of the site.
- The CSS isn't really very well written and there are numerous redundant rules and CSS variables just lying around everywhere. It was quite a frustrating experience working with the file. Considering our code quality standards for our other repos, I believe that a similar amount of rigor can be expected from this project.

I understand that we were under time constraints during the initial implementation. But IMO, now's a good opportunity to make this timeline _our own_ and do a serious refactor of it all before the next time someone comes around and have to do all the hard work updating the timeline in the future. 

---

<small><em>(note: the proposal was born out of a frustrating experience from working with this timeline for my dark theme PR. perhaps others don't see too much of a problem with it currently, but I hope to make life easier for whoever that might touch it later)</em></small>

### PR DESCRIPTION
**Pull request by hedyhli**
Closes #1235.

- Timeline is now its standalone app that reads the source files on startup
- Each entry now supports links (jump to entry) as a side-benefit of this feature
- Everything *should* look the same, ideally, further testing would be needed[^1]

[^1]: There are several instances where the ordering changed in the new version, this is because some items aren't currently correctly sorted by date. For example (left is before, right is after):

Preview: https://deploy-preview-1293--pydis-static.netlify.app/timeline/

The key used for sorting is a date in the "2006-01-02" format which comes from the filename. The latter part of the filename is used as anchor links.

The date specified in the frontmatter is the actual format used for display.

Dates shown in the timeline currently have a range of formats:
- Jan 2nd, 2006
- January 2nd, 2006
- January 2, 2006
  
All are copied as-is from the HTML.

I went ahead to convert all files to get things going quickly, since content needed to added sooner or later. Changing the format of YAML and file names **is still possible**, if reviewers insist.

<details><summary>Conversion data and scripts for reference</summary>

```nushell
#!/usr/bin/env nu
for x in ( cat entries.json | from json -s ) {
    let slug = ($x.title | str downcase | str replace -ar '[^ \w]' ' ' | str kebab-case)
    let filename = $"pydis_site/apps/timeline/entries/($x.date | into datetime | format date '%Y-%m-%d_')($slug).md"
    mut content = $"---
title: ($x.title)
date: ($x.date)
icon: ($x.icon)
"
    if ($x.icon_color | str length) > 0 {
        $content += $"icon_color: ($x.icon_color)"
    }
    $content += $"
---

($x.content | python3 -c 'print('\n'.join(__import__('textwrap').wrap(open(0).read(),width=80)))')
"
    $content | save $filename
    echo $slug
}
```

<details><summary>Recovered JSON</summary>

```json
[
  {
    "title": "Switch to new paste service",
    "date": "Jul 11th, 2023",
    "icon": "fa fa-comments",
    "icon_color": "pastel-pink",
    "content": "We migrate over to [pinnwand](https://github.com/supakeen/pinnwand)\nas the service that powers our paste bin over at\n[https://paste.pythondiscord.com/](https://paste.pythondiscord.com).\nWe made the switch as it comes with native light/dark modes, support for multi-file\npastes, additional support for text highlighting languages, and plus, it's written in\nPython!"
  },
  {
    "title": "Retirement of Joe and Sebastiaan",
    "date": "Jan 30th, 2023",
    "icon": "pydis",
    "icon_color": "",
    "content": "Having been at the helm of Python Discord for over 5 and 3 years respectively, Joe and\nSebastiaan retire and step down. They gain the @Founders role and continue as advisors\nto the @Directors, the new name of the original @Owners role.\n\nAt the same time, Mina and Zig join Leon as co-directors."
  },
  {
    "title": "Switch to forum-based help system",
    "date": "Nov 25th, 2022",
    "icon": "fa fa-comments",
    "icon_color": "pastel-pink",
    "content": "We migrate our help system to use a forum channel, retiring our home-grown rotating help\nsystem after 3 years of service and nearly 500,000 help sessions. Forum channels offer\na better experience as members can create their own dedicated thread in a discoverable place."
  },
  {
    "title": "Python 3.11 Release Stream",
    "date": "Oct 24th, 2022",
    "icon": "pydis",
    "icon_color": "",
    "content": "With the Python 3.10 Release Stream being such a success, we brought it back for the\nrelease of Python 3.11. Hosted by Leon, and CPython 3.11 Release Manager, Pablo Galindo,\nthey were joined by other CPython Core Developers. Together, they discuss the specific\nfeatures and the overall development process of the release.\n\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/PGZPSWZSkJI\"></iframe>"
  },
  {
    "title": "Summer Code Jam 2022 (CJ9)",
    "date": "July 21st, 2022",
    "icon": "fa fa-dice",
    "icon_color": "pastel-lime",
    "content": "We host the 9th Code Jam. This year, teams had to use **websockets** to create a\nproject based on the theme, **It's not a bug, it's a feature**. In all, 24 teams\nsubmitted their projects. At the end, we held a livestream demoing the top 10 projects\nand announcing the winner!\n\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/YZcVjFED6Hg\"></iframe>"
  },
  {
    "title": "Partnership with pyqtgraph",
    "date": "May 19th, 2022",
    "icon": "fa fa-handshake",
    "icon_color": "",
    "content": "The `#pyqtgraph` channel is created for the Scientific Graphics and GUI\nlibrary pyqtgraph, joining `#black-formatter`."
  },
  {
    "title": "Addition of @Sir Robin",
    "date": "Feb 21st, 2022",
    "icon": "fa fa-robot",
    "icon_color": "pastel-blue",
    "content": "Our arsenal of bots grows! We add @Sir Robin to power and manage all of our future\nevents and to support the Events Team."
  },
  {
    "title": "Trivia Night",
    "date": "Feb 12th, 2022",
    "icon": "fa fa-question",
    "icon_color": "pastel-green",
    "content": "How well do you know Python inside out? Members got to find out in a Trivia Night event.\nContestants were given questions about Python's internals, its development, and more.\nTo win, contestants had to get the most questions right while being fast to answer."
  },
  {
    "title": "Creation of Events Team",
    "date": "Feb 9th, 2022",
    "icon": "pydis",
    "icon_color": "",
    "content": "We form the Events Team to organise and run future events. Led by Kat and comprised by\nstaff members, the goal of the team is to ultimately host more events in a more\nsustainable way."
  },
  {
    "title": "Deployment of Smarter Resources",
    "date": "Feb 2nd, 2022",
    "icon": "fa fa-code",
    "icon_color": "pastel-red",
    "content": "We gave our resources pages some much needed love and\n[reorganised them into a single page](https://www.pythondiscord.com/resources/),\ncomplete with a shiny new resource filter that allows you to more quickly find\nones that relate to your interests, experience, learning style, and ability to pay!"
  },
  {
    "title": "We hit 300 000 members!",
    "date": "Jan 19, 2022",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "Thanks to an increasing growth rate, Python Discord's membership count doubled from\n150,000 to 300,000 in less than a year!"
  },
  {
    "title": "We host the Python 3.10 Release Stream",
    "date": "Oct 4th, 2021",
    "icon": "pydis",
    "icon_color": "",
    "content": "Leon and Pablo Galindo, CPython Core Developer and Release Manager, host the Python 3.10\nRelease Stream, joined by other core devs Carol Willing, Irit Katriel, \u0141ukasz Langa, and\nBrandt Bucher. They talked in-depth about the new features introduced in 3.10, the\ndevelopment of those features, and more!\n\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/AHT2l3hcIJg\"></iframe>"
  },
  {
    "title": "Partnership with Black",
    "date": "May 24th, 2021",
    "icon": "fa fa-handshake",
    "icon_color": "",
    "content": "We partner with the uncompromising code formatter project, Black, who were looking for a\nnew home for their real-time chat. Python Discord ended up being that home, resulting in\nthe creation of the `#black-formatter` channel."
  },
  {
    "title": "Summer Code Jam 2021 (CJ8)",
    "date": "July 9, 2021",
    "icon": "fa fa-dice",
    "icon_color": "pastel-orange",
    "content": "We host the 8th now-annual Code Jam. Teams had to create a program with an text-based user\ninterface (TUI), all designed around the theme of \u201cthink inside the box.\u201d Ultimately, 51\nteams submitted projects.\n[The winning submissions are listed on our website.](https://www.pythondiscord.com/events/code-jams/8/)"
  },
  {
    "title": "Inaugural run of Pixels",
    "date": "May 24, 2021",
    "icon": "fa fa-palette",
    "icon_color": "pastel-purple",
    "content": "Inspired by the subreddit, r/place, Pixels was our collaborative canvas event held\nbetween May 25 to June 14, providing a beginner-friendly API to paint pixels on a\nvirtual canvas.\n\n\nLater, we released [a blog post](https://blog.pythondiscord.com/pixels-summer-2021/)\nsummarizing what happened, our motives, and some stories from during development."
  },
  {
    "title": "Owners become PSF Fellows",
    "date": "April 23, 2021",
    "icon": "pydis",
    "icon_color": "",
    "content": "Joe, Leon, and Sebastiaan\n[are recognized as Python Software Foundation Fellows](https://pyfound.blogspot.com/2021/04/python-software-foundation-fellow.html) \nfor their substantial contributions to the Python ecosystem by fostering\nPython Discord."
  },
  {
    "title": "Summer Code Jam 2020 Highlights",
    "date": "Mar 21st, 2021",
    "icon": "fa fa-youtube-play",
    "icon_color": "pastel-red",
    "content": "We release a new video to our YouTube showing the best projects from the Summer Code Jam 2020.\nBetter late than never!\n\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/g9cnp4W0P54\"></iframe>"
  },
  {
    "title": "New feature: Weekly discussion channel",
    "date": "Mar 13th, 2021",
    "icon": "fa fa-comment",
    "icon_color": "pastel-purple",
    "content": "Every week (or two weeks), we'll be posting a new topic to discuss in a\nchannel called **#weekly-topic-discussion**. Our inaugural topic is a PyCon talk by Anthony Shaw called\n**Wily Python: Writing simpler and more maintainable Python.**.\n\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/dqdsNoApJ80\"></iframe>"
  },
  {
    "title": "We're on the Teaching Python podcast!",
    "date": "Mar 13th, 2021",
    "icon": "fa fa-microphone",
    "icon_color": "pastel-pink",
    "content": "Leon joins Sean and Kelly on the Teaching Python podcast to discuss how the pandemic has\nchanged the way we learn, and what role communities like Python Discord can play in this new world.\nYou can find the episode [at teachingpython.fm](https://www.teachingpython.fm/63)."
  },
  {
    "title": "Leon Sand\u00f8y appears on Talk Python To Me",
    "date": "Mar 1st, 2021",
    "icon": "fa fa-microphone",
    "icon_color": "pastel-green",
    "content": "Leon goes on the Talk Python to Me podcast with Michael Kennedy\nto discuss the history of Python Discord, the critical importance of culture, and how to run a massive\ncommunity. You can find the episode  [at talkpython.fm](https://talkpython.fm/episodes/show/305/python-community-at-python-discord)."
  },
  {
    "title": "We now have 150,000 members!",
    "date": "Feb 18th, 2021",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "Our growth continues to accelerate."
  },
  {
    "title": "We release The PEP 8 song",
    "date": "February 8th, 2021",
    "icon": "fa fa-music",
    "icon_color": "pastel-blue",
    "content": "We release the PEP 8 song on our YouTube channel, which finds tens of\nthousands of listeners!\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/hgI0p1zf31k\"></iframe>"
  },
  {
    "title": "Advent of Code attracts hundreds of participants",
    "date": "December 1st - 25th, 2020",
    "icon": "fa fa-snowflake-o",
    "icon_color": "pastel-red",
    "content": "A total of 443 Python Discord members sign up to be part of\n[Eric Wastl's excellent Advent of Code event](https://adventofcode.com/).\nAs always, we provide dedicated announcements, scoreboards, bot commands and channels for our members\nto enjoy the event in."
  },
  {
    "title": "We migrate all our infrastructure to Kubernetes",
    "date": "Nov 29th, 2020",
    "icon": "fa fa-wrench",
    "icon_color": "pastel-orange",
    "content": "As our tech stack grows, we decide to migrate all our services over to a\ncontainer orchestration paradigm via Kubernetes. This gives us better control and scalability.\n**Joe Banks** takes on the role as DevOps Lead."
  },
  {
    "title": "Python Discord hits 100,000 members!",
    "date": "Oct 22nd, 2020",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "Only six months after hitting 40,000 users, we hit 100,000 users. A\nmonumental milestone,\nand one we're very proud of. To commemorate it, we create this timeline."
  },
  {
    "title": "Python Discord hosts the 2020 CPython Core Developer Q&A",
    "date": "Oct 21st, 2020",
    "icon": "pydis",
    "icon_color": "",
    "content": "<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/gXMdfBTcOfQ\"></iframe>"
  },
  {
    "title": "Python Discord is now the new home of the PyWeek event!",
    "date": "Aug 16th, 2020",
    "icon": "fa fa-handshake",
    "icon_color": "pastel-pink",
    "content": "PyWeek, a game jam that has been running since 2005, joins Python\nDiscord as one of our official events. Find more information about PyWeek on [their official website](https://pyweek.org/)."
  },
  {
    "title": "PyDis summer code jam 2020 with the theme \u201cEarly Internet\u201d and Django as the technology",
    "date": "Jul 31st, 2020",
    "icon": "fa fa-dice",
    "icon_color": "pastel-blue",
    "content": "Sponsored by the Django Software Foundation and JetBrains, the Summer\nCode Jam for 2020 attracts hundreds of participants, and sees the creation of some fantastic\nprojects. Check them out in our judge stream below:\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/OFtm8f2iu6c\"></iframe>"
  },
  {
    "title": "Python Discord Public Statistics are now live",
    "date": "Jun 4th, 2020",
    "icon": "fa fa-chart-bar",
    "icon_color": "pastel-red",
    "content": "After getting numerous requests to publish beautiful data on member\ncount and channel use, we create [stats.pythondiscord.com](https://stats.pythondiscord.com/) for\nall to enjoy."
  },
  {
    "title": "Python Discord is now listed on python.org/community",
    "date": "May 28th, 2020",
    "icon": "fa fa-handshake",
    "icon_color": "pastel-pink",
    "content": "After working towards this goal for months, we finally work out an\narrangement with the PSF that allows us to be listed on that most holiest of websites:\nhttps://python.org/. [There was much rejoicing.](https://youtu.be/yciX2meIkXI?t=3)"
  },
  {
    "title": "ModMail is now live",
    "date": "May 25th, 2020",
    "icon": "fa fa-comments",
    "icon_color": "pastel-green",
    "content": "Having originally planned to write our own ModMail bot from scratch, we\ncome across an exceptionally good [ModMail bot by\nkyb3r](https://github.com/kyb3r/modmail) and decide to just self-host that one instead."
  },
  {
    "title": "PyDis Game Jam 2020 with the \u201cThree of a Kind\u201d theme and Arcade as the technology",
    "date": "Apr 17th, 2020",
    "icon": "fa fa-gamepad",
    "icon_color": "pastel-purple",
    "content": "The creator of Arcade, Paul Vincent Craven, joins us as a judge.\nSeveral of the Code Jam participants also end up getting involved contributing to the Arcade\nrepository.\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/KkLXMvKfEgs\"></iframe>"
  },
  {
    "title": "Python Discord hits 40,000 members, and is now bigger than Liechtenstein.",
    "date": "Apr 14, 2020",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "![](https://cdn.discordapp.com/attachments/354619224620138496/699666518476324954/unknown.png)"
  },
  {
    "title": "The new help channel system is live",
    "date": "Apr 5th, 2020",
    "icon": "fa fa-comments",
    "icon_color": "pastel-green",
    "content": "We release our dynamic help-channel system, which allows you to claim\nyour very own help channel instead of fighting over the static help channels. We release a [Help Channel Guide](https://pythondiscord.com/pages/resources/guides/help-channels/) to\nhelp our members fully understand how the system works."
  },
  {
    "title": "PyDis sixth code jam with the theme \u201cAncient technology\u201d and the technology Kivy",
    "date": "Jan 17, 2020",
    "icon": "fa fa-dice",
    "icon_color": "pastel-blue",
    "content": "Our Code Jams are becoming an increasingly big deal, and the Kivy core\ndevelopers join us to judge the event and help out our members during the event. One of them,\n@tshirtman, even joins our staff!\n\n<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/8fbZsGrqBzo\"></iframe>"
  },
  {
    "title": "PyDis hits 30,000 members",
    "date": "Dec 22nd, 2019",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "More than tripling in size since the year before, the community hits\n30000 users. At this point, we're probably the largest Python chat community on the planet."
  },
  {
    "title": "Sebastiaan Zeef becomes an owner",
    "date": "Sept 22nd, 2019",
    "icon": "pydis",
    "icon_color": "",
    "content": "After being a long time active contributor to our projects and the driving\nforce behind many of our events, Sebastiaan Zeef joins the Owners Team alongside Joe & Leon."
  },
  {
    "title": "The code of conduct is created",
    "date": "Oct 26th, 2019",
    "icon": "fa fa-scroll",
    "icon_color": "pastel-lime",
    "content": "Inspired by the Adafruit, Rust and Django communities, an essential\ncommunity pillar is created; Our [Code of\nConduct.](https://pythondiscord.com/pages/code-of-conduct/)"
  },
  {
    "title": "The Django rewrite of pythondiscord.com is now live!",
    "date": "Sep 15, 2019",
    "icon": "fa fa-code",
    "icon_color": "pastel-orange",
    "content": "The site is getting more and more complex, and it's time for a rewrite.\nWe decide to go for a different stack, and build a website based on Django, DRF, Bulma and\nPostgreSQL."
  },
  {
    "title": "PyDis hits 15,000 members; the \u201chot ones special\u201d video is released",
    "date": "Apr 8th, 2019",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "<iframe allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen=\"\" class=\"force-aspect-content\" frameborder=\"0\" src=\"https://www.youtube.com/embed/DIBXg8Qh7bA\"></iframe>"
  },
  {
    "title": "django-simple-bulma is released on PyPi",
    "date": "Dec 19th, 2018",
    "icon": "fa fa-code",
    "icon_color": "pastel-orange",
    "content": "Our very first package on PyPI, [django-simple-bulma](https://pypi.org/project/django-simple-bulma/) is a package that\nsets up the Bulma CSS framework for your Django application and lets you configure everything in\nsettings.py."
  },
  {
    "title": "PyDis hits 10,000 members",
    "date": "Nov 24th, 2018",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "We partner with RLBot, move from GitLab to GitHub, and start putting\ntogether the first Advent of Code event."
  },
  {
    "title": "First Hacktoberfest PyDis event; @Sir Lancebot is created",
    "date": "Oct 1st, 2018",
    "icon": "fa fa-dice",
    "icon_color": "pastel-blue",
    "content": "We create a second bot for our community and fill it up with simple,\nfun and relatively easy issues. The idea is to create an approachable arena for our members to cut\ntheir open-source teeth on, and to provide lots of help and hand-holding for those who get stuck.\nWe're training our members to be productive contributors in the open-source ecosystem."
  },
  {
    "title": "PyDis is now partnered with Discord; the vanity URL discord.gg/python is created",
    "date": "Jul 10th, 2018",
    "icon": "fa fa-handshake",
    "icon_color": "pastel-pink",
    "content": "After being rejected for their Partner program several times, we\nfinally get approved. The recent partnership with the r/Python subreddit plays a significant role in\nqualifying us for this partnership."
  },
  {
    "title": "PyDis hits 5,000 members and partners with r/Python",
    "date": "Jun 20th, 2018",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "As we continue to grow, we approach the r/Python subreddit and ask to\nbecome their official Discord community. They agree, and we become listed in their sidebar, giving\nus yet another source of new members."
  },
  {
    "title": "Do You Even Python and PyDis merger",
    "date": "Jun 9th, 2018",
    "icon": "fa fa-handshake",
    "icon_color": "pastel-pink",
    "content": "At this point in time, there are only two serious Python communities on\nDiscord - Ours, and one called Do You Even Python. We approach the owners of DYEP with a bold\nproposal - let's shut down their community, replace it with links to ours, and in return we will let\ntheir staff join our staff. This gives us a big boost in members, and eventually leads to @eivl and\n@Mr. Hemlock joining our Admin team"
  },
  {
    "title": "The privacy policy is created",
    "date": "May 21st, 2018",
    "icon": "fa fa-scroll",
    "icon_color": "pastel-lime",
    "content": "Since data privacy is quite important to us, we create a privacy page\npretty much as soon as our new bot and site starts collecting some data. To this day, we keep [our privacy policy](https://pythondiscord.com/pages/privacy/) up to date with all\nchanges, and since April 2020 we've started doing [monthly data reviews](https://pythondiscord.notion.site/6784e3a9752444e89d19e65fd4510d8d)."
  },
  {
    "title": "First code jam with the theme \u201csnakes\u201d",
    "date": "Mar 23rd, 2018",
    "icon": "fa fa-dice",
    "icon_color": "pastel-blue",
    "content": "Our very first Code Jam attracts a handful of users who work in random\nteams of 2. We ask our participants to write a snake-themed Discord bot. Most of the code written\nfor this jam still lives on in Sir Lancebot, and you can play with it by using the\n`.snakes` command. For more information on this event, see [the event page](https://pythondiscord.com/pages/code-jams/code-jam-1-snakes-bot/)"
  },
  {
    "title": "PyDis hits 2,000 members; pythondiscord.com and @Python are live",
    "date": "Mar 4th, 2018",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "The public moderation bot we're using at the time, Rowboat, announces\nit will be shutting down. We decide that we'll write our own bot to handle moderation, so that we\ncan have more control over its features. We also buy a domain and start making a website in Flask."
  },
  {
    "title": "Our logo is born. Thanks @Aperture!",
    "date": "Feb 3rd, 2018",
    "icon": "pydis",
    "icon_color": "",
    "content": "![](https://raw.githubusercontent.com/python-discord/branding/main/logos/logo_banner/logo_site_banner.svg)"
  },
  {
    "title": "Python Discord hits 1,000 members",
    "date": "Nov 10th, 2017",
    "icon": "fa fa-users",
    "icon_color": "pastel-dark-blue",
    "content": "Our main source of new users at this point is a post on Reddit that\nhappens to get very good SEO. We are one of the top 10 search engine hits for the search term\n\"python discord\"."
  },
  {
    "title": "Python Discord is created",
    "date": "Jan 8th, 2017",
    "icon": "pydis",
    "icon_color": "",
    "content": "**Joe Banks** becomes one of the owners around 3 days after it\nis created, and **Leon Sand\u00f8y** (lemon) joins the owner team later in the year, when the community\nhas around 300 members."
  }
]
```

Note that some iframes were stripped by markdownify.

</details></details>

### CODE REVIEW

**Comment by jchristgit** [comment]
This is great !!!!!!

**Comment by jchristgit** [comment]
This is still great !!!!!!!!!!!!!

File: pydis_site/apps/timeline/entries/2021-07-09_summer-code-jam-2021-cj8.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: Summer Code Jam 2021 (CJ8)
+date: July 9, 2021
+icon: fa fa-dice
+icon_color: pastel-orange
+---
+
+We host the 8th now-annual Code Jam. Teams had to create a program with an text-
+based user interface (TUI), all designed around the theme of “think inside the
```
Comment (wookie184) [inline pydis_site/apps/timeline/entries/2021-07-09_summer-code-jam-2021-cj8.md]: This adds an unwanted space.
Before:
image
After:
image

**Review by wookie184 — CHANGES_REQUESTED** [review]
This is great, only one issue I noticed.

I wonder if we should make the date metadata optional (to allow for date ranges) and extract it from the file name by default. This would help avoid the inconsistency of month names you mentioned

**Comment by hedyhli** [comment]
> I wonder if we should make the date metadata optional (to allow for date ranges) and extract it from the file name by default. This would help avoid the inconsistency of month names you mentioned

Good idea 👍

**Review by jchristgit — APPROVED** [review]
This is an excellent and super clean implementation - thank you very much!

**Review by wookie184 — APPROVED** [review]
We can leave making the date field optional and extracting from the file name to potentially another PR, this is great as-is so there's no point leaving it. Let's merge

**Pull request merged**